nupic/research/frameworks/vernon/mixins/extra_validations.py

```diff
@@ -0,0 +1,140 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy as np
+
+from nupic.research.frameworks.pytorch.model_utils import aggregate_eval_results
+
+__all__ = [
+    "ExtraValidations"
+]
+
+
+class ExtraValidations:
```

Comment:
I would suggest a more specific name like `ExtraValidationsPerEpoch` or something similar. `ExtraValidations` is quite generic, and I could only get its meaning by going through the docstring and code. By the name only I initially thought it was a way to add more performance measures to the validation process.